spharpy/classes/audio.py

```diff
@@ -0,0 +1,178 @@
+from pyfar import Signal
+from spharpy.spherical import renormalize, change_channel_convention
+import numpy as np
+
+
+class SphericalHarmonicSignal(Signal):
+    """Create audio object with spherical harmonics coefficients in time or
+    frequency domain.
+
+    Objects of this class contain spherical harmonics coefficients which are
+    directly convertible between time and frequency domain (equally spaced
+    samples and frequency bins), the channel conventions ACN and FUMA, as
+    well as the normalizations N3D, SN3D, or MaxN, see [#]_. The definition of
+    the spherical harmonics basis functions is based on the scipy convention
+    which includes the Condon-Shortley phase, [#]_, [#]_.
+
+
+    Parameters
+    ----------
+    data : ndarray, double
+        Raw data of the spherical harmonics signal in the time or
+        frequency domain. The data should have at least 3 dimensions,
+        according to the 'C' memory layout, e.g. data of
+        ``shape = (1, 4, 1024)`` has 1 channel with 4 spherical harmonic
+        coefficients with 1024 samples or frequency
+        bins each. Time data is converted to ``float``. Frequency is
+        converted to ``complex`` and must be provided as single
+        sided spectra, i.e., for all frequencies between 0 Hz and
+        half the sampling rate.
+    sampling_rate : double
+        Sampling rate in Hz
+    basis_type : str
+        Type of spherical harmonic basis, either ``'complex'`` or
+        ``'real'``.
+    normalization : str
+        Normalization convention, either ``'n3d'``, ``'maxN'`` or
+        ``'sn3d'``. (maxN is only supported up to 3rd order)
+    channel_convention : str
+        Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+        (FuMa is only supported up to 3rd order)
+    condon_shortley : bool or str, optional
+        Whether to include the Condon-Shortley phase term. If ``True``
+        Condon-Shortley is included, if ``False`` it is not included.
```

Comment:
This would be a bit more accurate for me:
```suggestion
        Flag to indicate if the Condon-Shortley phase term is included (``True``) or not (``False``).
```